tests/ui/proc-macro/quote/auxiliary/basic.rs

```diff
@@ -50,6 +53,23 @@ pub fn run_tests(_: TokenStream) -> TokenStream {
     TokenStream::new()
 }
 
+fn test_sep() {
+    let iter = ["a", "b"].into_iter();
+    let tokens = quote!($($iter) << *);
+
+    let expected = "\"a\" << \"b\"";
+    assert_eq!(expected, tokens.to_string());
+}
+
+fn test_dollar_dollar() {
+    let iter = ["a", "b"].into_iter();
+    let x = "X";
+    let tokens = quote!($$ x $($$ x $iter)*);
```

Comment:
This case is not actually fixed. It will fail to compile if you remove the unused definition of `x`.